tests/client/test_query.py

```diff
@@ -61,3 +61,132 @@ def test_get_container_sources():
         'upstream_urls': [yum_utils_upstream_url, etcd_upstream_url]
     }
     assert query.get_container_content_sources(container_build_id) == expected
+
+
+def test_get_current_and_previous_versions():
+    """Test the get_current_and_previous_versions function."""
+    go = Component(
+        canonical_name='golang', canonical_type='generic', canonical_namespace='redhat').save()
+    next_sl = None
+    url = 'git://pkgs.domain.local/rpms/golang?#fed96461b05c0078e537c93a3fe974e8b334{version}'
+    for version in ('1.9.7', '1.9.6', '1.9.5', '1.9.4', '1.9.3'):
+        sl = SourceLocation(
+            url=url.format(version=version.replace('.', '')), canonical_version=version).save()
+        sl.component.connect(go)
+        if next_sl:
+            next_sl.previous_version.connect(sl)
+        next_sl = sl
+
+    rv = query.get_current_and_previous_versions('golang', 'generic', '1.9.6')
+    versions = set([result['canonical_version'] for result in rv])
+    assert versions == set(['1.9.6', '1.9.5', '1.9.4', '1.9.3'])
+
+
+def test_get_container_built_with_artifact():
+    """Test the get_current_and_previous_versions function."""
```

Comment:
Wrong docstring.

Also, could do with more commentary here explaining the situation we're setting up for testing.